adapters/play/src/main/scala/caliban/Uploads.scala

```diff
@@ -0,0 +1,128 @@
+package caliban
+
+import java.nio.file.{ Files, Path }
+
+import caliban.InputValue.ListValue
+import caliban.Uploads.{ FileMeta, Uploads }
+import caliban.Value.{ NullValue, StringValue }
+import caliban.schema.Annotations.GQLName
+import zio.blocking.Blocking
+import zio.stream.{ Stream, ZSink, ZStream }
+import zio.{ Chunk, Has, Layer, RIO, UIO, URIO, ZIO, ZLayer }
+
+@GQLName("Upload")
+sealed case class Upload(name: String) {
+
+  val allBytes: RIO[Uploads with Blocking, Chunk[Byte]] =
+    Uploads.stream(name).run(ZSink.foldLeftChunks(Chunk[Byte]())(_ ++ (_: Chunk[Byte])))
+
+  val meta: URIO[Uploads, Option[FileMeta]] =
+    Uploads.fileMeta(name)
+
+}
+
+object Uploads {
+  type Uploads = Has[Service.Service]
```

Comment:
I think the typical pattern is for this to be a package object with `Uploads` under it, so then it would be used like 

```scala
import caliban.uploads._

Uploads.empty
```

or 

```scala
uploads.stream("filename")
```

This seems slightly odd that it is `Service.Service`